### ISSUE TITLE
div by zero

### ISSUE THREAD

**Issue by CorBer** [TAdeJong/weighed_phase_unwrap#1]
Hi Tobias,

Bij toeval liep ik tegen jouw code aan, ik ben wat aan het experimenteren met interferograms en zocht een unwrapper. Die van jou werkt erg goed ! Wel liep ik in phase_unwrap op regel 93 soms tegen div by zero errors aan. Als een van de scale elementen op nul gaat krijg je die foutmelding. Ik heb zelf op regel 167 de volgende regel ingevoegd:  scaling[scaling == 0] = 0.00000001

groet
Cor Berrevoets

**Comment by TAdeJong** [comment]
Hi Cor,

(Continuing in English for readability for anyone international that may come across this issue)

Thank you for your bug report! I had noticed this behaviour, but did not have time to fix it.
The divide by zero was a remnant of the different handling by the original matlab code, which solved this by explicitly filling in the element afterwards. I now updated the code to handle it more gracefully. For the reference version, I did this by adding the `with np.errstate(divide='ignore'):` context handler, to keep it as close as possible to the matlab implementation.
For the optimized version, I decided to follow the original paper by Ghiglia et al., and set the scaling of the [0,0] component, which is the only one causing trouble here, to 1., to keep the bias of the picture the same.

Als je nog verdere vragen heb hoor ik het graag!
